library/alloc/src/slice.rs

```diff
@@ -511,9 +511,9 @@ impl<T> [T] {
             while m > 0 {
                 // `buf.extend(buf)`:
                 unsafe {
-                    ptr::copy_nonoverlapping(
+                    ptr::copy_nonoverlapping::<T>(
                         buf.as_ptr(),
-                        (buf.as_mut_ptr() as *mut T).add(buf.len()),
+                        (buf.as_mut_ptr()).add(buf.len()),
```

Comment:
Turbofish required here, otherwise
```
error[E0699]: cannot call a method on a raw pointer with an unknown pointee type
   --> library\alloc\src\slice.rs:516:44
    |
516 |                         (buf.as_mut_ptr()).add(buf.len()),
    |                                            ^^^

error[E0699]: cannot call a method on a raw pointer with an unknown pointee type
   --> library\alloc\src\slice.rs:535:75
    |
535 |                 ptr::copy_nonoverlapping(buf.as_ptr(), (buf.as_mut_ptr()).add(buf.len()), rem_len);
    | 
```